### DISCUSSION TITLE
New license and move to other repository/organisation

### DISCUSSION THREAD

**Discussion by necropola** [matrix-org/dendrite#3281] [Help & Support] [answered]
Reading https://element.io/blog/element-to-adopt-agplv3/ (from 2023-11-06) and in particular
> For those running unmodified free-standing open source instances of [Synapse](https://github.com/matrix-org/synapse), [Dendrite](https://github.com/matrix-org/dendrite), [Sygnal](https://github.com/matrix-org/sygnal), [Sydent](https://github.com/matrix-org/sydent) and [MAS](https://github.com/matrix-org/matrix-authentication-service), there is no change beyond needing to point your deployments at the new repositories in the Element rather than Matrix GitHub organization. These will be available in the coming days.

I wonder what the current state and plans are for dendrite.

**Comment by necropola** [comment]
`matrix-org/synapse` has now been forked to `element-hq/synapse` and has then been archived at its original place. In doing so they have lost links to all existing issues (https://github.com/element-hq/synapse/issues/6) and they now require signing a CLA by anyone wanting to contribute (to a PR). What does that mean for **dendrite**? 
 
**2023-12-13:** https://element.io/blog/synapse-now-lives-at-github-com-element-hq-synapse/
 > As [we announced in November](https://element.io/blog/element-to-adopt-agplv3/), Element is switching to use the Affero General Public License (AGPLv3) for its future contributions to Synapse and related backend Matrix projects.  The repositories will live at [github.com/element-hq](https://github.com/element-hq) (newly renamed from our old outdated name of github.com/vector-im); we’re switching them over starting today, commencing with Synapse.
 > ...
 > In order to sell AGPL exceptions, Element needs to have permission from future code contributors to be able to redistribute their contributions under a non-AGPL licence. This means that future contributors will need to sign a Contributor Licence Agreement (CLA) to grant Element the permission to distribute their code under non-AGPL terms.
 > ...

**Reply by S7evinK** [reply, marked as answer]
The idea is basically the same for Dendrite - once Synapse has been forked/moved successfully, the same is going to happen to Dendrite. As you have linked, there has been some trouble around issues etc., once those are resolved and everything looks fine, Dendrite will undergo the same procedure. (This also means we're changing the license and require a signed CLA after the change.)

**Reply by necropola** [reply]
Hmm, but doesn't the CLA (agree upfront to commercial non-AGPL License exceptions) deter the same group of possible contributors/maintainers that the original Apache License did?

I'll mark this as answered, but I was really hoping for `dendrite` to go with a pure AGPL License without exceptions (just like the [mautrix/whatsapp](https://github.com/mautrix/whatsapp), [mautrix/signal](https://github.com/mautrix/signal), ... bridges) to see, whether this attracts more/enough contributors/maintainers. Especially since **polylith** was dropped and small personal installations are  more likely the target than big distributed enterprise deployments.

What, if someone creates a public fork of the then AGPL licensed repository and keeps it synced as close as possible? Contributations to that fork wouldn't be subject to the CLA used by the source, right?
